### PR TITLE
Add fill_time and fill_value dataset props

### PR DESCRIPTION
**Pull request by mkitti**
This adds dataset creation properties for `fill_time` and `fill_value`.

### CODE REVIEW

File: src/api_midlevel.jl [hunk 1]
```diff
@@ -185,3 +185,17 @@ function write_chunk(dataset_id, index::Integer, buf::Union{DenseArray,Base.Fast
     offset = [reverse(get_chunk_offset(dataset_id, index))...]
     write_chunk(dataset_id, offset, buf; dxpl_id = dxpl_id, filter_mask = filter_mask)
 end
+
+function get_fill_value(plist_id, ::Type{T}) where T
```
Comment (musm) [inline src/api_midlevel.jl:189]: Would the second argument ever not be `Float64`?

Comment (mkitti) [reply src/api_midlevel.jl:189]: One might want an `Int64` at some point.

Comment (mkitti) [reply src/api_midlevel.jl:189]: I defaulted to `Float64` since that should cover the most common cases.

Comment (musm) [reply src/api_midlevel.jl:189]: Ok I see that that is the case in the test examples. And it doesn't seem like something we could infer.

Comment (mkitti) [reply src/api_midlevel.jl:189]: Right. I cannot find a way to find the type unless we have a reference to the associated dataset.

**Comment by musm** [comment]
LGTM

**Comment by mkitti** [comment]
What's with CI these days?

**Comment by mkitti** [comment]
thanks

**Comment by musm** [comment]
At some point we should make an announcement on discourse with all these new v0.16 improvements 😄

**Comment by mkitti** [comment]
I'm thinking of at least one more on file free space strategy.
https://docs.hdfgroup.org/hdf5/rfc/FileSpaceManagement.pdf
https://docs.hdfgroup.org/archive/support/HDF5/docNewFeatures/NewFeaturesFileSpaceMgmtDocs.html
https://docs.hdfgroup.org/hdf5/develop/group___f_c_p_l.html#ga167ff65f392ca3b7f1933b1cee1b9f70

**Pull request merged**